awx/main/scheduler/task_manager.py

```diff
@@ -249,20 +272,47 @@ def timeout_approval_node(self):
     def get_tasks(self):
         workflow_jobs_running = [wf for wf in WorkflowJob.objects.filter(status='running')]
         workflow_jobs_pending = [wf for wf in WorkflowJob.objects.filter(status='pending')]
+        return workflow_jobs_running, workflow_jobs_pending
+
+    def start_pending_workflows(self, workflow_jobs_running, workflow_jobs_pending):
+        """Take in list of currently pending and running workflows.
+
+        Start the workflows that we can and send any notifications or websockets needed.
+        Return list of the now running workflows.
```

Comment:
Something isn't clicking for me here. We both saw that this problem is solved twice, here and by the dependency graph. What I don't understand is why you would keep _this_ logic, and remove workflows from the dependency_graph construct.

Maybe that's not clear enough... so re-framing...

Why should the `WorkflowManager` process "pending" workflows? Wouldn't this be better addressed in the main task manager, and have the workflow manager be specialized to spawning workflow jobs and processing finished workflows? The blocking logic for pending jobs is very much the domain of the main task manager.

Consider that sliced jobs probably don't respect project `update_on_launch` correctly. That introduces a dependency between pending workflow jobs and project updates.. or at least it would if we did it correctly. By having a separate class process pending workflow jobs, you're closing the door on fixing that competently.

Reply:
My reasoning was:
 - Do all workflow stuff in the WorkflowManager
 - update_on_launch behavior gets resolved when actual jobs of the sliced job workflow get created -- as you said, this is how it works now, nothing in current logic considers this. To me, that makes sense. Workflows are just a container for actual jobs. No need to update_on_launch to start the container, the inner nodes are the ones that matter.
 - only dependency ever considered in the DependencyGraph was allow_simultaneous
 - We were not build the DependencyGraph in this class so I retained the allow_simultaneous logic that was previously scattered through process_pending_jobs and start_task
 
My main goal was to have  a KISS solution where only the WorkflowManager managed workflows. I'm not beyond convincing.